### PR TITLE
Default Parameter Value blog post

### PR DESCRIPTION
**Pull request by doctordns**
# PR Summary
A new post on $PSDefaultParameterValues variable. 

## PR Checklist

- Do you want this post to be published on a specific date?: Y/N [Specify date]:
- [x] I have read the [contributors guide][contrib] and followed the style and process guidelines
- [x] PR has a meaningful title
- [ ] Includes content related an open issue
  - see [Closing issues using keywords][key].
- [x] This PR is ready to merge and is not **Work in Progress**

[contrib]: https://docs.microsoft.com/powershell/scripting/community/contributing/overview
[key]: https://help.github.com/en/articles/closing-issues-using-keywords

### CODE REVIEW

File: Posts/2021/12/tfl-params.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+---
+post_title: How to Use $PSDefaultParameterValues
+username: tfl@psp.co.uk
+Categories: PowerShell
+tags: PowerShell, Default Parameter values, parameters
+Summary: Using The $PSDefaultParameterValues automatic variable
+---
+
+**Q:** When I use cmdlets like `Receive-Job` and `Format-Table`, how do I change default values of **-Keep** and **-Wrap**?
```
Comment (sdwheeler) [inline Posts/2021/12/tfl-params.md:9]:
```suggestion
**Q:** When I use cmdlets like `Receive-Job` and `Format-Table`, how do I change default values of the **Keep** and **Wrap** parameters?
```

File: Posts/2021/12/tfl-params.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+**A:** Use the `$PSDefaultValues` automatic variable.
+
+When I first discovered PowerShell's background jobs feature, I would use `Receive-Job` to view job output - only to discover it's no longer there.
+And almost too often to count, I pipe objects to `Format-Table` cmdlet only to get truncated output because I forgot to use `-Wrap`.
+I'm sure you all have parameters whose default value you would gladly change - at least for your environment.
+
+I'm sure you have seen this (and know how to use -Wrap), like this:
```
Comment (sdwheeler) [inline Posts/2021/12/tfl-params.md:17]:
```suggestion
I'm sure you have seen this (and know how to use **Wrap**), like this:
```

File: Posts/2021/12/tfl-params.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                        for example configuration history and
+                        Application Pool account mapping. If this
+                        service is stopped, configuration history
+                        and locking down files or directories with
+                        Application Pool specific Access Control
+                        Entries will not work.
+
+```
+
+So, the question is: how to tell PowerShell to always use -Wrap when using `Format-Table` or `Format-List`.
```
Comment (sdwheeler) [inline Posts/2021/12/tfl-params.md:53]:
```suggestion
So, the question is: how to tell PowerShell to always use `-Wrap` when using `Format-Table` or `Format-List`?
```

File: Posts/2021/12/tfl-params.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+It turns out there is a very simple way: use the `$PSDefaultParameters` automatic variable.
+
+## The **$PSDefaultParameters** automatic variable
```
Comment (sdwheeler) [inline Posts/2021/12/tfl-params.md:56]:
```suggestion
## The `$PSDefaultParameters` automatic variable
```

File: Posts/2021/12/tfl-params.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+When Powershell (and Windows PowerShell ) starts, it creates the `$PSDefaultParameters' automatic variable.
```
Comment (sdwheeler) [inline Posts/2021/12/tfl-params.md:58]:
```suggestion
When PowerShell (and Windows PowerShell) starts, it creates the `$PSDefaultParameters` automatic variable.
```

File: Posts/2021/12/tfl-params.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+The variable has a type: **System.Management.Automation.DefaultParameterDictionary**.
+In other words, the variable is a Powershell hash table.
+By default the variable is empty when you start PowerShell.
```
Comment (sdwheeler) [inline Posts/2021/12/tfl-params.md:61]:
```suggestion
By default, the variable is empty when you start PowerShell.
```

File: Posts/2021/12/tfl-params.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+Each entry in this hash table defines a cmdlet, a parameter and a default value for that parameter.
+The hash table key is the name of the hash table, followed by a ":", and then the name of the parameter.
```
Comment (sdwheeler) [inline Posts/2021/12/tfl-params.md:64]:
```suggestion
The hash table key is the name of the cmdlet, followed by a colon (`:`), and then the name of the parameter.
```

File: Posts/2021/12/tfl-params.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+The hash table value for this key is the new default value for the parameter.
+
+If you wanted, for example, to always use **-Wrap** for the `Format-*` cmdlets, you could do this:
```
Comment (sdwheeler) [inline Posts/2021/12/tfl-params.md:67]:
```suggestion
If you wanted, for example, to always use the **Wrap** parameter for the `Format-*` cmdlets, you could do this:
```

File: Posts/2021/12/tfl-params.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+```PowerShell
+$PSDefaultParameterValues.Add('Format-*:Wrap', $True)
+```
+## Persist the change in your profile
+Any change you make to the **$PSDefaultParameterValues** variable is only applicable for the current session.
```
Comment (sdwheeler) [inline Posts/2021/12/tfl-params.md:73]:
```suggestion
Any change you make to the `$PSDefaultParameterValues` variable is only applicable for the current session.
```

File: Posts/2021/12/tfl-params.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+And the variable is subject to normal scoping rules - so changing the value in a script does not affect the session as a whole.
+That means that if you want these changes to occur every time you start a PowerShell console, then you add the appropriate statements in your profile.
+
+On my development box, I use the following snippet in side my **$Profile** file:
```
Comment (sdwheeler) [inline Posts/2021/12/tfl-params.md:77]:
```suggestion
On my development box, I use the following snippet inside my `$PROFILE` script:
```

File: Posts/2021/12/tfl-params.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+
+```powerShell
+$PSDefaultParameterValues.Add('*:AutoSize', $true)
+$PSDefaultParameterValues.Add('Receive-Job:Keep', $true)
+$PSDefaultParameterValues.Add('*:Wrap', $true
```
Comment (sdwheeler) [inline Posts/2021/12/tfl-params.md:82]: I think you meant for those parameters to apply to the `Format-*` cmdlets.

```suggestion
$PSDefaultParameterValues.Add('Format-*:AutoSize', $true)
$PSDefaultParameterValues.Add('Format-*:Wrap', $true)
$PSDefaultParameterValues.Add('Receive-Job:Keep', $true)
```

File: Posts/2021/12/tfl-params.md [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+```
+
+## Summary
+
+The `$PSDefaultParameterValues` automatic variable is a great tool to help you specify specific values for cmdlet parameters.
+You can specify one or more cmdlets by using wild cards in the hash table's key.
+Remember that the hash table key is the name of the cmdlet(s), a colon, and then the parameter's name.
+Also, the hash table value is the new "default" value for that parameter (and for the specified cmdlet(s)).
+
+You can read more about `$PSDefaultParameterValues`, and other preference variables in [about_Preference_Variables](https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_preference_variables#psdefaultparametervalues). And for more details of parameter default values, see the [about_Parameters_Default_Values help file](https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_parameters_default_values).
```
Comment (sdwheeler) [inline Posts/2021/12/tfl-params.md:92]:
```suggestion
You can read more about `$PSDefaultParameterValues`, and other preference variables in [about_Preference_Variables](https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_preference_variables#psdefaultparametervalues). 
And for more details of parameter default values, see the [about_Parameters_Default_Values help file](https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_parameters_default_values).

```

**Review by sdwheeler — CHANGES_REQUESTED** [review]
See my suggestions inline.

I noticed a lot of inconsistent formatting. Please review the style guide again.

**Comment by doctordns** [comment]
It has been a very bad few weeks here - apologies for it taking so long to resolve these.

I think I have fixed these and this is therefore ready to go.

**Review by sdwheeler — APPROVED** [review]

**Pull request merged**